omc3/optics_measurements/data_models.py

```diff
@@ -74,17 +77,25 @@ def dpps(self, plane: str) -> np.ndarray:
         return np.array([df.DPP for df in self[plane]])
 
     def dpp_frames(self, plane, dpp_value):
-        dpp_dfs = []
-        for i in np.argwhere(np.abs(self.dpps(plane) - dpp_value) < 1e-6).T[0]:
-            dpp_dfs.append(self[plane][i])
+        if dpp_value is None:
+            return self._all_frames(plane)
+
+        dpp_dfs = [self[plane][i] for i in self.dpp_frames_indices(plane, dpp_value)]
         if len(dpp_dfs) == 0:
             raise ValueError(f"No data found for dp/p {dpp}")
         return dpp_dfs
+    
+    def dpp_frames_indices(self, plane, dpp_value):
+        """ Return the indices of the frames that match the dpp. """
+        if dpp_value is None:
+            return list(range(len(self[plane])))
+
+        return np.argwhere(np.abs(self.dpps(plane) - dpp_value) < DPP_TOLERANCE).T[0]
 
     def _all_frames(self, plane):
         return self[plane]
 
-    def joined_frame(self, plane, columns, dpp_value=None, dpp_amp=False, how='inner'):
+    def joined_frame(self, plane, columns, dpp_value=None, dpp_amp=False, how='inner', dtype = np.float64):
```

Comment:
Please type hints, let's add as we go :(

Reply:
I don't know about putting a default value to the dtype parameter. If we ever call this function on a dframe with complex data, the entire imaginary part will be discarded and we won't be aware.

I'd rather leave it as `None` and make sure we specify it wherever we call `joined_frame`

Reply:
I think it will be a long time before we use complex columns in our data output. And even longer before this will become an issue. And even then, the complex data will  be the exception, where we can then actually use this parameter.